features/unit/abijson.feature

```diff
@@ -0,0 +1,75 @@
+@unit.abijson
+@unit
+Feature: AbiJson
+  Scenario Outline: Serialize Method object from sig into json 
+    When I create the Method object from "<methodsig>"
+    And I serialize the Method object into json
+    Then the produced json should equal "<jsonfile>" loaded from "<directory>"
+
+    Examples:
+      | methodsig                | jsonfile       | directory         |
+      | add(uint32,uint32)uint32 | addMethod.json | abi_responsejsons |
+
+  Scenario Outline: Create and serialize Method object into json
+    When I create the Method object with "<name>" "<firstargtype>" "<secondargtype>" "<returntype>"
+    And I serialize the Method object into json
+    Then the produced json should equal "<jsonfile>" loaded from "<directory>"
+
+    Examples:
+      | name  | firstargtype | secondargtype | returntype | jsonfile       | directory         |
+      | add   | uint32       | uint32        | uint32     | addMethod.json | abi_responsejsons |
+
+  Scenario Outline: Create and serialize Method object into json with arg names
+    When I create the Method object with "<name>" "<firstargname>" "<firstargtype>" "<secondargname>" "<secondargtype>" "<returntype>"
+    And I serialize the Method object into json
+    Then the produced json should equal "<jsonfile>" loaded from "<directory>"
+
+    Examples:
+      | name  | firstargname | firstargtype | secondargname | secondargtype | returntype | jsonfile                   | directory         |
+      | add   | first        | uint32       | second        | uint32        | uint32     | addMethodWithArgNames.json | abi_responsejsons |
+
+  Scenario Outline: Create and serialize Method object into json with description
+    When I create the Method object with "<name>" "<methoddesc>" "<firstargtype>" "<firstdesc>" "<secondargtype>" "<seconddesc>" "<returntype>"
+    And I serialize the Method object into json
+    Then the produced json should equal "<jsonfile>" loaded from "<directory>"
+
+    Examples:
+      | name  | methoddesc  | firstargtype | firstdesc   | secondargtype | seconddesc  | returntype | jsonfile                      | directory         |
+      | add   | description | uint32       | description | uint32        | description | uint32     | addMethodWithDescription.json | abi_responsejsons |
```

Comment:
You could increase the strength of these tests by not using repeated values for the parameters. This specific test uses `uint32` for all types and `description` for all description, but a stronger test would vary these values to guard against a buggy implementation mixing them up.